jax/_src/pjit.py

```diff
@@ -374,9 +368,48 @@ def cache_miss(*args, **kwargs):
 
   cpp_pjitted_f = wraps(fun)(cpp_pjit_f)
   cpp_pjitted_f._fun = fun
-  type(cpp_pjitted_f).clear_cache = _cpp_pjit_evict_fn
+  cpp_pjitted_f._jit_info = jit_info
```

Comment:
FWIW, this seems weird to add `jit_info` on the class and then access it in trace, lower, eval_shape. Is there a different way of achieving it?

Reply:
The `_jit_info` should not be on the class, it is an attribute of the function. It is needed because other methods need it. This is the standard way to set attributes that methods need. We cannot close over the `_pjit_info` before we assign to the class, because the methods assigned on the class need to be constant.